src/services/accounts/AccountsStakingPayoutsService.ts

```diff
@@ -533,29 +540,56 @@ export class AccountsStakingPayoutsService extends AbstractService {
 			era: EraIndex,
 			stakers: KeysAndExposures,
 			validatorIndex: ValidatorIndex,
+			validatorsOverviewEntries?: [StorageKey, Option<SpStakingPagedExposureMetadata>][],
 		): IAdjustedDeriveEraExposure {
 			const nominators: DeriveEraNominatorExposure = {};
 			const validators: DeriveEraValidatorExposure = {};
+			const validatorsOverview: Record<string, Option<SpStakingPagedExposureMetadata>> = {};
 
 			stakers.forEach(([key, exposure]): void => {
 				const validatorId = key.args[1].toString();
 
+				if (validatorsOverviewEntries) {
+					for (const validator of validatorsOverviewEntries) {
+						const validatorKey: StorageKey = validator[0];
+						const valKey: [string, string] = validatorKey.toHuman() as unknown as [string, string];
+						if (valKey) {
+							if (valKey[1].toString() === validatorId) {
+								validatorsOverview[validatorId] = validator[1];
+								break;
+							}
+						}
+					}
+				}
+
 				validators[validatorId] = exposure;
 
-				exposure.others.forEach(({ who }, validatorIndex): void => {
-					const nominatorId = who.toString();
+				if (exposure.others) {
```

Comment:
Since the same logic around `who` is repeated here in both cases perhaps we can do something like

```
				const individualExposure = exposure.others ? exposure.others : (exposure as unknown as Option<SpStakingExposurePage>).unwrap().others;
				individualExposure.forEach(({ who }, validatorIndex): void => {
					const nominatorId = who.toString();

					nominators[nominatorId] = nominators[nominatorId] || [];
					nominators[nominatorId].push({ validatorId, validatorIndex });
				});
```

Reply:
Awesome! Changed it! Thank you for the suggestion! 💯

Reply:
Accounting for Tarik's suggestion below this could look something like

```
const individualExposure = exposure.others ? exposure.others : 
(exposure as unknown as Option<SpStakingExposurePage>).isSome ?
(exposure as unknown as Option<SpStakingExposurePage>).unwrap().others :
undefined;

if (individualExposure) {
	individualExposure.forEach(({ who }, validatorIndex): void => {
		const nominatorId = who.toString();

		nominators[nominatorId] = nominators[nominatorId] || [];
		nominators[nominatorId].push({ validatorId, validatorIndex });
	});
}
```

Reply:
I already changed this.

I did it like this 

```
const individualExposure = exposure.others
  ? exposure.others
  : (exposure as unknown as Option<SpStakingExposurePage>).isSome
	  ? (exposure as unknown as Option<SpStakingExposurePage>).unwrap().others
	  : [];
  individualExposure.forEach(({ who }, validatorIndex): void => {
  const nominatorId = who.toString();
  
  nominators[nominatorId] = nominators[nominatorId] || [];
  nominators[nominatorId].push({ validatorId, validatorIndex });
  });
```

it looks the same as yours but instead of `undefined` i set an empty array. Then forEach in an empty array will just not do anything. 
Is it recommended to set `undefined` and then check `if (individualExposure) ` or I can leave it like that ?

Reply:
nope, this is totally cool! great job